assets/admin/students/student-modal/index.test.js

```diff
@@ -1,94 +1,290 @@
+import nock from 'nock';
 /**
  * Internal dependencies
  */
 import { StudentModal } from './index';
 
-const coursePromise = Promise.resolve( [
+const courses = [
 	{
 		id: 1,
-		title: { rendered: 'My Course' },
+		title: { rendered: 'Course 1' },
+	},
+	{
+		id: 2,
+		title: { rendered: 'Course 2' },
+	},
+	{
+		id: 3,
+		title: { rendered: 'Course 3' },
 	},
-] );
+];
 
-jest.mock( '@wordpress/api-fetch', () => jest.fn() );
-apiFetch.mockImplementation( () => coursePromise );
+const students = [ 1, 2, 3 ];
+const NONCE = 'some-nounce-id';
 
 describe( '<StudentModal />', () => {
+	const { getByText, getByRole, findByText, findByRole } = screen;
+
+	beforeAll( () => {
+		nock( 'http://localhost' )
+			.persist()
+			.get( '/wp-json/wp/v2/courses' )
+			.query( { per_page: 100 } )
+			.reply( 200, courses );
+
+		nock( 'http://localhost' )
+			.persist()
+			.get( '/wp-admin/admin-ajax.php' )
+			.query( { action: 'rest-nonce' } )
+			.reply( 200, NONCE );
+	} );
+	afterAll( () => nock.cleanAll() );
+
+	it( 'Should display a list of courses', async () => {
+		render( <StudentModal action="add" /> );
+		expect(
+			await findByText( courses.at( 0 ).title.rendered )
+		).toBeTruthy();
+	} );
+
+	it( 'Should disable the action button when there is no course selected', async () => {
+		render( <StudentModal action="add" /> );
+		expect(
+			getByRole( 'button', { name: 'Add to Course' } )
+		).toBeDisabled();
+	} );
+
 	describe( 'Add action', () => {
-		beforeEach( async () => {
-			await act( async () => {
-				return render( <StudentModal action="add" /> );
-			} );
+		const onClose = jest.fn();
+
+		beforeEach( () => {
+			render(
+				<StudentModal
+					action="add"
+					onClose={ onClose }
+					students={ students }
+				/>
+			);
 		} );
 
 		it( 'Should display the action description', async () => {
 			expect(
-				screen.getByText(
+				getByText(
 					'Select the course(s) you would like to add students to:'
 				)
 			).toBeTruthy();
 		} );
 
 		it( 'Should display the action button', async () => {
 			expect(
-				screen.getByRole( 'button', { name: 'Add to Course' } )
+				getByRole( 'button', { name: 'Add to Course' } )
 			).toBeTruthy();
 		} );
+
+		it( 'Should add the selected students to the selected course', async () => {
+			nock( 'http://localhost' )
+				.post( '/', {
+					student_ids: students,
+					course_ids: [ courses.at( 0 ).id ],
+				} )
+				.query( {
+					rest_route: '/sensei-internal/v1/course-students/batch',
+					_wpnonce: NONCE,
+				} )
+				.once()
+				.reply( 200 );
+
+			fireEvent.click(
+				await screen.findByLabelText( courses.at( 0 ).title.rendered )
+			);
+
+			fireEvent.click(
+				await findByRole( 'button', { name: 'Add to Course' } )
+			);
+			await waitFor( () => {
+				expect( onClose ).toHaveBeenCalledWith( true );
+			} );
+		} );
+
+		describe( 'when there is a failure to add the students to the courses', () => {
```

Comment:
As per my earlier comment, does this need to be a separate describe? Separating it by actions makes sense, but this feels like an unnecessary level.

Reply:
I would like to put it in a separate context because we have other scenarios that we should check during the error. 
For example, Now I am going to add a scenario to test if the button is disabled after an error, the error that you got.

Reply:
I moved the scenario's test setup to a beforeEach and I added a new scenario only for this context, so now I think the `describe` usage does make more sense.
What do you think?